components/collector/src/source_collectors/github/merge_requests.py

```diff
@@ -0,0 +1,148 @@
+        pagination = f'(first: 100, after: "{cursor}")'
+        project = f"{self._parameter('owner')}/{self._parameter('repository')}"
+        pull_request_query = PULL_REQUEST_QUERY.format(pagination=pagination)
+        response = await client.execute(
+            pull_request_query,
+            variables={
+                "repository": self._parameter("repository"),
+                "owner": self._parameter("owner"),
+            },
+        )
+        json = await response.json()
+        if repository := json["data"]["repository"]:
+            page_info = repository["pullRequests"]["pageInfo"]
+            return response, page_info["hasNextPage"], page_info.get("endCursor", "")
+        raise GitHubPullRequestInfoError(project)
+
+    async def _parse_entities(self, responses: SourceResponses) -> Entities:
+        """Override to parse the pull requests."""
+        pull_requests = []
+        for response in responses:
+            json = await response.json()
+            pull_requests.extend(json["data"]["repository"]["pullRequests"]["nodes"])
+        return Entities(self._create_entity(pr) for pr in pull_requests)
+
+    async def _parse_total(self, responses: SourceResponses) -> Value:
+        """Override to parse the total number of pull requests."""
+        return str((await responses[0].json())["data"]["repository"]["pullRequests"]["totalCount"])
+
+    def _create_entity(self, pull_request) -> Entity:
+        """Create an entity from a GitHub JSON pull request."""
+        return Entity(
+            key=pull_request["id"],
+            title=pull_request["title"],
+            base_ref_name=pull_request["baseRefName"],
+            url=pull_request["url"],
+            state=pull_request["state"],
+            review_decision=self.__review_decision_state(pull_request["reviewDecision"]),
+            created=pull_request["createdAt"],
+            updated=pull_request["updatedAt"],
+            merged=pull_request["mergedAt"],
+            comments=str(pull_request["comments"]["totalCount"]),
+            thumbs_up=str(pull_request["reactions"]["totalCount"]),
+        )
+
+    def _include_entity(self, entity: Entity) -> bool:
+        """Return whether the pull request should be counted."""
+        pr_matches_state = entity["state"] in self._parameter("merge_request_state")
+        pr_matches_review_decision = entity["review_decision"] in self._parameter("review_decision")
+        base_ref_name = entity["base_ref_name"]
+        branches = self._parameter("target_branches_to_include")
+        pr_matches_branches = match_string_or_regular_expression(base_ref_name, branches) if branches else True
+
+        return pr_matches_state and pr_matches_branches and pr_matches_review_decision
+
+    @classmethod
+    def __review_decision_state(cls, review_decision) -> str:
+        if isinstance(review_decision, str):
+            return review_decision
+        return "?"
```

Comment:
```suggestion
        return review_decision if isinstance(review_decision, str) else "?"
```

Also, please add missing docstring explaining why we need to check for a string type.